packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart

```diff
@@ -320,3 +320,70 @@ class AndroidJavaScriptChannelParams extends JavaScriptChannelParams {
 
   final android_webview.JavaScriptChannel _javaScriptChannel;
 }
+
+/// Object specifying creation parameters for creating a [AndroidWebViewWidget].
+///
+/// When adding additional fields make sure they can be null or have a default
+/// value to avoid breaking changes. See [PlatformWebViewWidgetCreationParams] for
+/// more information.
+@immutable
+class AndroidWebViewWidgetCreationParams
+    extends PlatformWebViewWidgetCreationParams {
+  /// Creates [AndroidWebWidgetCreationParams].
+  AndroidWebViewWidgetCreationParams({
+    super.key,
+    required super.controller,
+    super.layoutDirection,
+    super.gestureRecognizers,
+    @visibleForTesting InstanceManager? instanceManager,
+  }) : instanceManager = instanceManager ?? JavaObject.globalInstanceManager;
+
+  /// Constructs a [WebKitWebViewWidgetCreationParams] using a
+  /// [PlatformWebViewWidgetCreationParams].
+  AndroidWebViewWidgetCreationParams.fromPlatformWebViewWidgetCreationParams(
+    PlatformWebViewWidgetCreationParams params, {
+    InstanceManager? instanceManager,
+  }) : this(
+          key: params.key,
+          controller: params.controller,
+          layoutDirection: params.layoutDirection,
+          gestureRecognizers: params.gestureRecognizers,
+          instanceManager: instanceManager,
+        );
+
+  /// Maintains instances used to communicate with the native objects they
+  /// represent.
+  ///
+  /// This field is exposed for testing purposes only and should not be used
+  /// outside of tests.
+  @visibleForTesting
+  final InstanceManager instanceManager;
+}
+
+/// An implementation of [PlatformWebViewWidget] with the Android WebView API.
+class AndroidWebViewWidget extends PlatformWebViewWidget {
+  /// Constructs a [WebKitWebViewWidget].
+  AndroidWebViewWidget(PlatformWebViewWidgetCreationParams params)
+      : super.implementation(
+          params is AndroidWebViewWidgetCreationParams
+              ? params
+              : AndroidWebViewWidgetCreationParams
+                  .fromPlatformWebViewWidgetCreationParams(params),
+        );
+
+  AndroidWebViewWidgetCreationParams get _androidParams =>
+      params as AndroidWebViewWidgetCreationParams;
+
+  @override
+  Widget build(BuildContext context) {
+    return AndroidView(
```

Comment:
I think this still uses VirtualDisplay. This should use the TLHC implementation with `PlatformViewLink` and `PlatformViewsService.initSurfaceAndroidView`:
https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart#L47